pkg/policy/condition/binaryequalsfunc.go

```diff
@@ -51,7 +52,20 @@ func (f binaryEqualsFunc) evaluate(values map[string][]string) bool {
 		requestValue = values[f.k.Name()]
 	}
 
-	return !f.values.Intersection(set.CreateStringSet(requestValue...)).IsEmpty()
+	fvalues := f.values.ApplyFunc(func(v string) string {
+		for _, key := range CommonKeys {
+			if strings.Contains(v, key.VarName()) {
+				kv := values[key.Name()][0]
+				if kv == "" { // Empty values are not supported for policy variables.
+					continue
+				}
+				return strings.Replace(v, key.VarName(), kv, -1)
```

Comment:
Multiple variables in a value is not replaced with values here e.g. `${aws:username}/${aws:SourceIp}` value contains two variables but either one value gets replaced.   I think below logic addresses all issues correctly.

```go
	fvalues := f.values.ApplyFunc(func(v string) string {
		for _, key := range CommonKeys {
			if rvalues, ok := values[key.VarName()]; ok && rvalues[0] != "" {
				v = strings.Replace(v, key.VarName(), rvalues[0], -1)
			}
		}

		return v
	})
```